mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2156,27 +1963,18 @@ createCPUConvFunc(ModuleOp module,
   }
   Type workspaceArgType;
   if (hasWorkspace) {
-    workspaceArgType = MemRefType::get(
-        ArrayRef<int64_t>(filterDimension.begin(), filterDimension.end()),
-        b.getF32Type());
+    workspaceArgType = MemRefType::get(filterElems, b.getF32Type());
   }
-
-  SmallVector<Type, 3> funcArgTypes = {filterType, inputType, outputType};
-
+  SmallVector<Type, 4> funcArgTypes = {filterType, inputType, outputType};
   if (hasWorkspace) {
-    funcArgTypes = {filterType, inputType, outputType, workspaceArgType};
+    funcArgTypes.push_back(workspaceArgType);
   }
 
   func =
       func::FuncOp::create(loc, funcName, b.getFunctionType(funcArgTypes, {}));
   module.push_back(func);
 
-  if (genValidation.getValue() == "mlir") { // -pv_with_mlir or -prc
-    createCPUConvWithMLIR(module, func, genConfig);
-  } else { // -pv_with_cpp
-    createCPUConvWithCPP(module, func, genConfig);
```

Comment:
Is this related with the change? I mean the fact that we don't have structured memrefs allowed us to remove `createCPUConvWithCPP` (and a lot of other code)?

Reply:
PR description updated